### ISSUE TITLE
`Yarn-tools` changes order of `integrity` field, moves it to tail

### ISSUE THREAD

**Issue by kompot** [atlassian/yarn-dedupe#8]
Hi,
with the recent addition of integrity field
https://github.com/yarnpkg/yarn/pull/5042
`yarn-tools` started to alter valid yarn.lock and it produce diffs with no real change.

Before running it is

```
"@types/react-virtualized@^9.7.4":
  version "9.18.5"
  resolved "https://registry.yarnpkg.com/@types/react-virtualized/-/react-virtualized-9.18.5.tgz#8c6b4e739e2fc4a601dd3e5e114dd0deeba56cc7"
  integrity sha512-ptqFDzemkXGMf7ylch/bCV+XTDvVjD9dRymzcjOPIxg8Hqt/uesOye10GXItFbsxJx9VZeJBYrR8FFTauu+hHg==
  dependencies:
    "@types/prop-types" "*"
    "@types/react" "*"
```

and after `yarn-tools` it is

```
"@types/react-virtualized@^9.7.4":
  version "9.18.5"
  resolved "https://registry.yarnpkg.com/@types/react-virtualized/-/react-virtualized-9.18.5.tgz#8c6b4e739e2fc4a601dd3e5e114dd0deeba56cc7"
  dependencies:
    "@types/prop-types" "*"
    "@types/react" "*"
  integrity sha512-ptqFDzemkXGMf7ylch/bCV+XTDvVjD9dRymzcjOPIxg8Hqt/uesOye10GXItFbsxJx9VZeJBYrR8FFTauu+hHg==
```

**Comment by lukebatchelor** [comment]
Related: https://github.com/yarnpkg/yarn/issues/6364

It's fixed in @yarnpkg/lockfile 1.1. I'll try

**Comment by Bnaya** [comment]
friendly ping @lukebatchelor  :)

**Comment by lukebatchelor** [comment]
The PR is up, just waiting on a merge. 

CC: @sergio

**Comment by scinos** [comment]
Fixed in `1.0-dev`. I'll close this issue once the patch is released as part of version 1.0
